src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -314,6 +345,27 @@ class ClusterTaskManager : public ClusterTaskManagerInterface {
   absl::flat_hash_map<SchedulingClass, std::deque<std::shared_ptr<internal::Work>>>
       tasks_to_schedule_;
 
+  /// Tracking information about the currently running tasks in a scheduling
+  /// class. This information is used to place a cap on the number of running
+  /// running tasks per scheduling class.
+  struct SchedulingClassInfo {
+    SchedulingClassInfo(int64_t cap)
+        : running_tasks(),
+          capacity(cap),
+          next_update_time(std::numeric_limits<int64_t>::max()) {}
```

Comment:
I think the logic will be simpler if you use 0 to mean nil instead of int_max.

Reply:
How so? What do you imagine the simplified logic would look like? I thought we would need a `if (get_time < wait_time) ...` case no matter what in order to handle the case where a task finishes before the timeout gets triggered (i.e. the last portion of `SchedulingClassCapIncrease` where we let one task finish)